zanata-war/src/main/java/org/zanata/dao/ProjectDAO.java

```diff
@@ -405,37 +409,36 @@ public HPerson getLastTranslator(HProject project) {
     public List<HProject> getProjectsForMaintainer(HPerson maintainer,
             String filter, int firstResult, int maxResults) {
         final String sqlFilter = filter == null ? "" : filter;
-        Query q =
-                getSession()
-                        .createQuery(
-                                "from HProject p " +
-                                "where :maintainer in elements(p.maintainers) " +
-                                "and p.status <> " +
-                                "'" + EntityStatus.OBSOLETE.getInitial() + "' " +
-                                "and (p.name like :filter " +
-                                    "or p.slug like :filter) " +
-                                "order by p.name")
-                        .setParameter("maintainer", maintainer)
-                        .setParameter("filter", "%" + sqlFilter + "%")
-                        .setFirstResult(firstResult)
-                        .setMaxResults(maxResults);
+        Query q = getSession().createQuery(
+                "select m.project from HProjectMember m " +
+                        "where m.person = :maintainer " +
+                        "and m.role = :role " +
+                        "and m.project.status <> :obsolete " +
+                        "and (m.project.name like :filter " +
+                        "or m.project.slug like :filter) " +
+                        "order by m.project.name")
+                .setParameter("maintainer", maintainer)
+                .setParameter("role", ProjectRole.Maintainer)
+                .setParameter("obsolete", EntityStatus.OBSOLETE)
+                .setParameter("filter", "%" + sqlFilter + "%")
```

Comment:
It does not seem desirable to me. It looks like the old query has the same behaviour, so we should make a bug if the behaviour should change.